tasks/task03/word_stats.py

```diff
@@ -0,0 +1,23 @@
+import re
+
+texts = [
+   "Hello, World!",
+   "The world is mine",
+   "Hello, how are you?"
+]
+
+words =[]
+
+for i, sentence in enumerate(texts):
+    index = i
+    for w in re.sub("[^\w]", " ", str.lower(sentence)).split():
+        dict_index = next((i for i, item in enumerate(words) if item['word'] == w), None)
+        if dict_index is None:
+            words.append({'word':w, 'count':1, 'index':i})
+
+        else:
+            words[dict_index]['count'] = words[dict_index]['count'] + 1
+
```

Comment:
In your code, you store statistics by word. So word is a key. It is much more performant to check word stats in a dict by key rather than searching through a list of words